torch/nested/_internal/nested_tensor.py

```diff
@@ -5,31 +5,32 @@
 import torch
 from torch._C import DispatchKey, DispatchKeySet
 from torch._prims_common import is_expandable_to
-from torch.nested._internal.nested_int import NestedIntNode
-from torch.utils.weak import WeakTensorKeyDictionary
+from torch.nested._internal.cached_tensor import CachedTensor
+from torch.nested._internal.nested_int import get_metadata, NestedIntNode
+from torch.nested._internal.tensor_registry import register_tensor, try_get_int
+from torch.nested._internal.utils import _try_get_fake_mode
 
 
-_tensor_id_counter = 0
-_tensor_symint_registry = WeakTensorKeyDictionary()
+def update_tensor_registry_hook(host_tensor, device_tensor):
+    register_tensor(host_tensor, try_get_int(device_tensor))
 
 
-def get_tensor_symint(tensor, *, coeff=1):
-    from torch._subclasses.fake_tensor import FakeTensor
-    from torch._subclasses.functional_tensor import mb_unwrap_functional_tensor
+# Fully represents raggedness
+SOURCE_FIELDS = ("_device_lengths", "_device_offsets", "_host_lengths", "_host_offsets")
+# Derived fields
+EXTRA_FIELDS = (
+    "_max_seqlen_tensor",
+    "_min_seqlen_tensor",
+    "_inverse_indices",
+)
+# UnpackResult = namedtuple("UnpackResult", SOURCE_FIELDS + EXTRA_FIELDS)
```

Comment:
need this or no?

Reply:
Whoops, will removed

Reminder to self to also remove `update_tensor_registry_hook`